adam/perception/perception_graph.py

```diff
@@ -501,7 +505,44 @@ def to_dot_node(pattern_node: "NodePredicate") -> str:
         dot_graph.render(output_file)
 
 
-@attrs(frozen=True, slots=True, eq=False)
+class DebugRenderingObject:
+    """
+        Helper callable class for debugging purposes. We render graph every hundred calls to match. The outputs can be
+        found in test/renders folder. This callable is called within GraphMatching.match
+    """
+
+    def __init__(self, graph_to_match_against: PerceptionGraph) -> None:
```

Comment:
Why do we need to provide `graph_to_match_against` to the constructor? Shouldn't we receive what graph information we need in `__call__`?